src/java.desktop/macosx/classes/sun/java2d/macos/MacOSFlags.java

```diff
@@ -44,43 +52,114 @@
 
     private static native boolean initNativeFlags();
 
-    private static boolean getBooleanProp(String p, boolean defaultVal) {
+    private static PropertyState getBooleanProp(String p, PropertyState defaultVal) {
         String propString = System.getProperty(p);
-        boolean returnVal = defaultVal;
+        PropertyState returnVal = defaultVal;
         if (propString != null) {
             if (propString.equals("true") ||
                 propString.equals("t") ||
                 propString.equals("True") ||
                 propString.equals("T") ||
                 propString.equals("")) // having the prop name alone
             {                          // is equivalent to true
-                returnVal = true;
+                returnVal = PropertyState.ENABLED;
             } else if (propString.equals("false") ||
                        propString.equals("f") ||
                        propString.equals("False") ||
                        propString.equals("F"))
             {
-                returnVal = false;
+                returnVal = PropertyState.DISABLED;
             }
         }
         return returnVal;
     }
 
-
-    private static boolean getPropertySet(String p) {
+    private static boolean isBooleanPropTrueVerbose(String p) {
         String propString = System.getProperty(p);
-        return (propString != null) ? true : false;
+        if (propString != null) {
+            if (propString.equals("True") ||
+                propString.equals("T"))
+            {
+                return true;
+            }
+        }
+        return false;
     }
 
     private static void initJavaFlags() {
         java.security.AccessController.doPrivileged(
                 (PrivilegedAction<Object>) () -> {
-                    metalEnabled = getBooleanProp("sun.java2d.metal", false);
+                    PropertyState oglState = getBooleanProp("sun.java2d.opengl", PropertyState.UNSPECIFIED);
+                    PropertyState metalState = getBooleanProp("sun.java2d.metal", PropertyState.UNSPECIFIED);
+
```

Comment:
Right here we could add
                    if (metalState == PropertyState.UNSPECIFIED && oglState == PropertyState.UNSPECIFIED) {
                        // change this line to set metalState to toggle the default pipeline.
                        oglState = PropertyState.ENABLED;
                    }
This makes it easier to set the default